packages/block-library/src/image/image.js

```diff
@@ -745,20 +745,24 @@ export default function Image( {
 							resetLightbox={ resetLightbox }
 						/>
 					) }
-					{ allowCrop && (
-						<ToolbarButton
-							onClick={ () => setIsEditingImage( true ) }
-							icon={ crop }
-							label={ __( 'Crop' ) }
-						/>
-					) }
-					{ showCoverControls && (
-						<ToolbarButton
-							icon={ overlayText }
-							label={ __( 'Add text over image' ) }
-							onClick={ switchToCover }
-						/>
-					) }
+					{ allowCrop &&
+						! isContentOnlyMode &&
+						hasNonContentControls && (
```

Comment:
Rather than adding a new condition here, I think it would be better to update the condition of the `allowCrop` variable. So the condition for the allowCrop variable should be:

```javascript
	const allowCrop =
		isSingleSelected &&
		canEditImage &&
		! isEditingImage &&
		hasNonContentControls;
```